File: src/main/java/it/contrader/services/LinkTKService.java
package it.contrader.services;

import java.util.ArrayList;
import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import ch.qos.logback.core.pattern.Converter;
import it.contrader.converter.ConverterKeyword;
import it.contrader.converter.ConverterLinkTK;
import it.contrader.converter.ConverterThing;
import it.contrader.dao.LinkTKRepository;

import it.contrader.dto.ThingDTO;
import it.contrader.dto.KeywordDTO;
import it.contrader.dto.LinkTKDTO;

import it.contrader.model.LinkTK;
import it.contrader.model.Keyword;
import it.contrader.model.Thing;


@Service
public class LinkTKService {
	private final LinkTKRepository linkTKRepository;
	
	@Autowired
	public LinkTKService(LinkTKRepository linkTKRepository) {
		this.linkTKRepository = linkTKRepository;
	}
	
	public List<LinkTKDTO> getListLinkTKDTO() {
		return ConverterLinkTK.toListDTO ((List<LinkTK>)linkTKRepository.findAll());
	}
	
	public LinkTKDTO insertLinkTK(LinkTKDTO linkTKDTO) {
		LinkTK link = ConverterLinkTK.toEntity(linkTKDTO);
		linkTKRepository.save(link); 
		return ConverterLinkTK.toDTO(link);
	}
	
	public boolean updateLinkTK(LinkTKDTO linkTKDTO) {
		return linkTKRepository.save(ConverterLinkTK.toEntity(linkTKDTO)) != null;
	}
	
	public boolean deleteLinkTK(int id) {
		this.linkTKRepository.deleteById(id);
		return true;
	}
	
	
	public List<LinkTKDTO> getAllByKeyword(KeywordDTO key){
		return ConverterLinkTK.toListDTO((List<LinkTK>) linkTKRepository.findAllByKeyword(ConverterKeyword.toEntity(key)));
	}
	
	public List<LinkTKDTO> getAllByThing(Thing thing){
		return ConverterLinkTK.toListDTO((List<LinkTK>) linkTKRepository.findAllByThing(thing));
	}
	
	
	/*
	public LinkTKDTO getLinkTKDTOById(int id) {
		return ConverterLinkTK.toDTO(linkTKRepository.findLinkTKByIdLinkTK(id));
	}*/
}

File: IncAng/src/app/services/linkTk.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { LinkTKDTO } from '../../dto/LinkTkDTO';
import { KeywordDTO } from '../../dto/KeywordDTO';

@Injectable({
  providedIn: 'root'
})
export class LinkTKService {
    private linkTKList: Array<LinkTKDTO>;
  
    constructor(private http: HttpClient) { }

    insertLinkTK(linkTKDTO: LinkTKDTO){
      console.log(linkTKDTO);
      return this.http.post('http://localhost:8080/LinkTK/insertLinkTK', linkTKDTO);
    }

    showThingOfKey(keywordDTO: KeywordDTO){
      return this.http.get('http://localhost:8080/Keyword/showThingOfKey?id='+ keywordDTO.idKeyword);
    }
}